libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/ModelOverview.tsx

```diff
@@ -167,6 +175,26 @@ export class ModelOverview extends React.Component<
         };
       });
 
+    let chartCohorts: ErrorCohort[];
+    let someCohortSelected: boolean;
+    let selectedChartCohorts: number[];
+    let labeledStatistics: ILabeledStatistic[][];
+    if (this.state.datasetCohortChartIsVisible) {
+      chartCohorts = this.context.errorCohorts;
+      someCohortSelected =
+        this.state.selectedDatasetCohorts !== undefined &&
+        this.state.selectedDatasetCohorts.length > 0;
+      selectedChartCohorts = this.state.selectedDatasetCohorts ?? [];
+      labeledStatistics = datasetCohortLabeledStatistics;
+    } else {
+      chartCohorts = featureBasedCohorts;
```

Comment:
nit: the if/else look similar, maybe refactor to a small method with different parameters?

Reply:
Yes, they're similar because I'm setting the same parameters that are needed for rendering. There's no logic that happens in the if, just setting of the parameter, so I don't quite know what a method would help with there. I could create a new component (not sure what I could call it since it is basically the ModelOverview) that takes in the view-agnostic params we're setting in this if/else, but then this component would just consist of this if/else and call the other component which doesn't seem that helpful. I will definitely look into breaking it up further once it's finished, but while there are still lots of pieces moving around I've found it to be easier to handle if it's not broken up into too many components.

Reply:
closing this since I'll do more refactoring later on (out of scope for this PR since the overall feature isn't finished)